nri/domain.go

```diff
@@ -0,0 +1,109 @@
+package nri
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/cri-o/cri-o/internal/log"
+	"github.com/sirupsen/logrus"
+
+	nri "github.com/containerd/nri/pkg/adaptation"
+)
+
+// Domain implements the functions the generic NRI interface needs to
+// deal with pods and containers from a particular containerd namespace.
```

Comment:
cri-o doesn't have a notion of  containerd namespaces, right? or does this PR introduce it and I haven't found it yet

Reply:
Yes, the comment is off... needs updating. domain even as an unexported concept here is probably one of the most debatable implementation details. I gutted it out compared to the original code (which really had support for more than a single registered domain), but it's still there. If that is a show-stopper we can get rid of it. 

That said, I tried (maybe a bit too desperately) to keep the NRI integration bits here very similar to how it's done in containerd. Basically split up to two pieces: an 'egress/downstream' interface (these bits here) and the pod+container-lifecycle hook-in bits (server/nri-api.go et. al). There are two reasons. One is that it makes it a bit easier for us to maintain both as the mental burden of switching between these is smaller. The other is that I hope that eventually we could most of the NRI 'egress/downstream' bits into common code in the NRI repo itself and just import it here.

Reply:
Updated those comments.